invoices/invoiceregistry.go

```diff
@@ -147,21 +147,36 @@ func NewRegistry(cdb *channeldb.DB, expiryWatcher *InvoiceExpiryWatcher,
 	}
 }
 
-// populateExpiryWatcher fetches all active invoices and their corresponding
-// payment hashes from ChannelDB and adds them to the expiry watcher.
-func (i *InvoiceRegistry) populateExpiryWatcher() error {
-	pendingOnly := true
-	pendingInvoices, err := i.cdb.FetchAllInvoicesWithPaymentHash(pendingOnly)
-	if err != nil && err != channeldb.ErrNoInvoicesCreated {
-		log.Errorf(
-			"Error while prefetching active invoices from the database: %v", err,
-		)
+// scanInvoicesOnStart will scan all invoices on start and add active invoices
+// to the invoice expirt watcher.
+func (i *InvoiceRegistry) scanInvoicesOnStart() error {
+	var (
+		pending  map[lntypes.Hash]*channeldb.Invoice
+		currScan int
+	)
+
+	scanFunc := func(scanCount int,
+		paymentHash lntypes.Hash, invoice *channeldb.Invoice) error {
+		if currScan != scanCount {
```

Comment:
👍

Reply:
Could use a comment explaining that this is us detecting if the entire transaction needed to be retried.

Reply:
I wonder if there's a simple abstraction we can deploy so this type of behavior becomes standardized for all calls to DB transactions.

Reply:
Added a comment, PTAL. Perhaps we could expose the retry count on the `kvdb` transaction level? That way there's no need to pass the count. Requires modification in the `walletdb` part though. Time to fork? :)